bitcode/devicelib.cl

```diff
@@ -772,6 +772,25 @@ EXPORT float CL_NAME2(shfl_down, f)(float var, uint delta) {
 
 
 
+EXPORT int CL_NAME2(max, i)(int a, int b) { return max(a, b); }
```

Comment:
OK, will move them

Reply:
I have moved the hip-side declarations into `int_math.hh`... but it's unclear to me where to move the OpenCL-side definitions. The only meaningful commit on `devicelib.cl` is the commit `sycn_and_util.hh`, and that doesn't do much with the OpenCL-side (e.g. `group_ballot` is declared but i don't see the definition anywhere; `local_barrier` is still defined in `devicelib.cl`). I assume this is still WIP ? in any case, you can go ahead and add commits to this PR.

Reply:
We should only add the complicated mangling in `devicelib.cl` for functions where the binding from HIP to OpenCL is not trivial. I want to remove the use of `GEN_NAME2` macros.

````
> In conclusion, it would seem that we can do all of the binding in spirv_hip_mathlib.hh without the need for double binding in mathlib.cl ?

Probably not all of it, but almost all of it. Looking at the mathlib.hh top comment:

 * 2) some functions have the same name in HIP as in OpenCL but different
 *    signature
 * 3) some OpenCL functions (e.g. geometric) take vector arguments
 *    but HIP/CUDA do not have vectors.

 these will likely still need an indirect call.
````

Reply:
fixed